fix: Fix X-axis flipping issue

Quanta-Naut opened the pull request:
I discovered issue where the rendered name appeared flipped and also the entire x-axis being flipped. This was caused by the x-axis being inverted during rendering. I’ve made the necessary adjustments to correct this behaviour.

The Last PR had conflicts so closed that, and now created new PR, to clean up.

Quanta-Naut commented:
Ahhhh, that would be a better way to test it, will do that.

Quanta-Naut commented:
@seveibar the test is failing because the font is different in the test environment, took a while to figure that out, let me know if I can fix it. Thanks.

seveibar commented:
basically there's a bug somewhere in the code with how the orientations are originally being created. Instead of fixing that, you're just flipping them all (which is inefficient and confusing!)

Quanta-Naut commented:
<p align="center">
  <img src="https://github.com/user-attachments/assets/b8498c14-06fb-4194-86f9-00e4e6ca4ce0" alt="Top" width="45%">
  <img src="https://github.com/user-attachments/assets/53ba587c-ae0b-4a8f-b729-05ee5e574d98" alt="Bottom" width="45%">
  <br>
  <b>Top</b> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <b>Bottom</b>
</p>

Quanta-Naut commented:
@seveibar sorry for the delay, had my exams going on. You were right about the bug, I removed the extra layer of transform and focused on fixing the bug, and also added the top and bottom view of a sample circuit.

seveibar commented:
@Quanta-Naut i investigated flipping the texture in https://github.com/tscircuit/circuit-json-to-gltf/pull/69/files

It doesn't work- you're correct that the X axis needs to be flipped but your placement of that flipping is not correct. But the good news is your approach is correct

Quanta-Naut commented:
@seveibar Thank you for the confirmation. I have a doubt tho, by wrong placement you mean withing the geometry.ts file or the bug must be fixed in a different file??

seveibar commented:
@Quanta-Naut this was patched yesterday!

You can look at the PR here: #70 

/tip $20
